VideoWeb/VideoWeb/ClientApp/src/app/shared/test-language.service.ts

```diff
@@ -0,0 +1,58 @@
+import { Injectable } from '@angular/core';
+import { TranslateService } from '@ngx-translate/core';
+import { filter } from 'rxjs/operators';
+
+@Injectable()
+export class TestLanguageService {
+    translationLanguageLoaded = false;
+    loggerPrefix = '[TestLanguageService] -';
+
+    constructor(private translateService: TranslateService) {}
+
+    setupSubscriptions() {
+        console.info(`${this.loggerPrefix} Subscribing to onLanguageChange`);
+        this.translateService.onLangChange.pipe(filter(e => e.lang === 'tl')).subscribe(() => this.setupTestLanguageTranslations());
+    }
+
+    setupTestLanguageTranslations() {
+        if (!this.translationLanguageLoaded) {
+            this.translationLanguageLoaded = true;
+            this.translateService
+                .getTranslation('en')
+                .toPromise()
```

Comment:
doesn't this work with observables?

Reply:
covered with tests so we can try :)

Reply:
the previous switch didnt with with tests, filter is working 🤷